### PR TITLE
fix(file_processors): allow owner-encrypted PDFs in pypdf processor

### LINKED ISSUE #6181: pypdf file processor rejects owner-encrypted PDFs that pypdf can parse
### System Info

- ogx latest (main branch)
- Python 3.12+
- pypdf 6.10+

### Information

- [x] My own modified scripts

### 🐛 Describe the bug

The `inline::pypdf` file processor in `src/ogx/providers/inline/file_processor/pypdf/pypdf.py` rejects all PDFs where `PdfReader.is_encrypted` returns `True`, including **owner-encrypted PDFs** that require no password to read.

Many corporate and financial documents (e.g., SEC filings, earnings press releases) use PDF owner-encryption to restrict printing/copying, but the content is freely readable without any password. `pypdf` handles these files correctly — `PdfReader` can extract text from them without calling `decrypt()`.

The current check at line 101:

```python
if reader.is_encrypted:
    raise HTTPException(status_code=422, detail="Password-protected PDFs are not supported")
```

This is too broad. `is_encrypted` returns `True` for **any** PDF with encryption metadata, not just password-protected ones. This blocks a common class of publicly available documents from being ingested into vector stores.

### Reproduction

1. Download any corporate PDF with owner-encryption (e.g., IBM quarterly earnings press release from https://www.ibm.com/investor)
2. Upload it to a vector store via the files API
3. The server rejects it with `422: Password-protected PDFs are not supported`
4. But running `PdfReader(file).pages[0].extract_text()` in Python works fine — the content is readable

### Expected behavior

The file processor should attempt to read the PDF before rejecting it. Owner-encrypted PDFs (no password needed) should be processed normally. Only PDFs that actually require a password to read should be rejected.

### Suggested fix

Replace the blanket rejection with an attempt to read:

```python
if reader.is_encrypted:
    try:
        reader.decrypt("")  # empty password works for owner-encrypted PDFs
    except Exception:
        raise HTTPException(status_code=422, detail="Password-protected PDFs are not supported")
```

This preserves the rejection for truly password-protected PDFs while allowing owner-encrypted documents through.

### PR DESCRIPTION
**Pull request by Artemon-line**
## Summary

The `is_encrypted` check in the pypdf file processor rejects all PDFs with any encryption metadata, including owner-encrypted documents that require no password to read. Many corporate PDFs (SEC filings, earnings press releases) use owner encryption to restrict printing/copying while keeping content freely readable.

This PR attempts `reader.decrypt("")` and checks the return value — this succeeds for owner-encrypted PDFs (`PasswordType.OWNER_PASSWORD = 2`) and returns `PasswordType.NOT_DECRYPTED = 0` for truly password-protected ones.

### Why check the return value (not just catch exceptions)

`pypdf`'s `decrypt()` uses a return-code pattern, not exceptions:

| `decrypt("")` result | `PasswordType` | int | bool | Meaning |
|---|---|---|---|---|
| Owner-encrypted (no user password) | `OWNER_PASSWORD` | `2` | `True` | Readable — allow |
| User password matches `""` | `USER_PASSWORD` | `1` | `True` | Readable — allow |
| Wrong password | `NOT_DECRYPTED` | `0` | `False` | Cannot read — reject 422 |
| Corrupted encryption metadata | raises `Exception` | — | — | Cannot read — reject 422 |

## Test Plan

```bash
uv run pytest tests/unit/providers/file_processor/test_pypdf_validation.py -v
```

### Condition coverage

Three new tests cover every branch of the `if reader.is_encrypted` block:

| Test | `decrypt("")` behavior | Expected | Verified against unfixed code |
|---|---|---|---|
| `test_rejects_password_protected_pdf` | returns `0` (`NOT_DECRYPTED`) | 422 | FAILS without fix (falls through) |
| `test_rejects_corrupted_encrypted_pdf` | raises `Exception` | 422 | passes (caught by `except`) |
| `test_allows_owner_encrypted_pdf` | returns `2` (`OWNER_PASSWORD`) | processes normally | passes (decrypt succeeds) |

```
tests/unit/providers/file_processor/test_pypdf_validation.py::test_rejects_docx_with_422 PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_rejects_pptx_with_422 PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_rejects_xlsx_with_422 PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_allows_pdf PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_allows_text_files PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_allows_csv_files PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_allows_markdown_files PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_rejects_password_protected_pdf PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_rejects_corrupted_encrypted_pdf PASSED
tests/unit/providers/file_processor/test_pypdf_validation.py::test_allows_owner_encrypted_pdf PASSED
10 passed
```

Fixes #6181

Signed-off-by: Artemy Hladenko <ahladenk@redhat.com>

### CODE REVIEW

File: src/ogx/providers/inline/file_processor/pypdf/pypdf.py [hunk 1]
```diff
@@ -103,7 +103,10 @@ def _process_pdf(
         reader = PdfReader(pdf_bytes)
 
         if reader.is_encrypted:
-            raise HTTPException(status_code=422, detail="Password-protected PDFs are not supported")
+            try:
+                reader.decrypt("")
```
Comment (skamenan7) [inline src/ogx/providers/inline/file_processor/pypdf/pypdf.py]: `reader.decrypt("")` returns `PasswordType.NOT_DECRYPTED` (0) for wrong passwords, it doesn't raise. A truly password-protected PDF will fall through here and produce a 500 or empty results. Can you check the return value? Something like `if not reader.decrypt(""):` would catch it. You may also need `except HTTPException: raise` above the broad `except Exception` so the 422 doesn't get swallowed.

File: tests/unit/providers/file_processor/test_pypdf_validation.py [hunk 2]
```diff
@@ -87,3 +88,39 @@ async def test_allows_markdown_files(pypdf_processor):
...
     assert len(result.chunks) >= 1
+
+
+async def test_rejects_password_protected_pdf(pypdf_processor):
+    pdf_bytes = b"%PDF-1.4\n1 0 obj\n<< /Type /Catalog /Pages 2 0 R >>\nendobj\n2 0 obj\n<< /Type /Pages /Kids [] /Count 0 >>\nendobj\nxref\n0 3\n0000000000 65535 f \n0000000009 00000 n \n0000000058 00000 n \ntrailer\n<< /Size 3 /Root 1 0 R >>\nstartxref\n115\n%%EOF"
+    file = UploadFile(filename="secret.pdf", file=io.BytesIO(pdf_bytes))
+
+    mock_reader = MagicMock(spec=PdfReader)
+    mock_reader.is_encrypted = True
+    mock_reader.decrypt.side_effect = Exception("password required")
```
Comment (skamenan7) [inline tests/unit/providers/file_processor/test_pypdf_validation.py]: This mocks `side_effect=Exception(...)` but real pypdf returns 0 for wrong passwords, doesn't raise. Once the return-value check is added, you would want a test with `mock_reader.decrypt.return_value = 0` to cover the common wrong-password path. The exception mock is still useful for corrupted-encryption edge cases.

File: src/ogx/providers/inline/file_processor/pypdf/pypdf.py [hunk 3]
```diff
@@ -103,7 +103,10 @@ def _process_pdf(
...
+            except Exception:
+                raise HTTPException(status_code=422, detail="Password-protected PDFs are not supported") from None
```
Comment (skamenan7) [inline src/ogx/providers/inline/file_processor/pypdf/pypdf.py:112]: Nice touch with `from None`,  keeps the 422 clean :)

**Review by skamenan7 — CHANGES_REQUESTED** [review]

File: src/ogx/providers/inline/file_processor/pypdf/pypdf.py (hunk 1 above)
Comment (Artemon-line) [reply src/ogx/providers/inline/file_processor/pypdf/pypdf.py]: Good catch — you're right, `decrypt("")` returns `PasswordType.NOT_DECRYPTED` (0) for wrong passwords instead of raising.

Fixed in 73700cb: now checking `if not reader.decrypt(""):` and re-raising `HTTPException` before the broad `except Exception`. All three branches covered by tests.

File: tests/unit/providers/file_processor/test_pypdf_validation.py (hunk 2 above)
Comment (Artemon-line) [reply tests/unit/providers/file_processor/test_pypdf_validation.py]: Done — split into two tests: `test_rejects_password_protected_pdf` with `return_value=0` for the common path, and `test_rejects_corrupted_encrypted_pdf` with `side_effect=Exception` for the edge case. Also changed the owner-encrypted mock to `return_value=2` (`OWNER_PASSWORD`).

File: src/ogx/providers/inline/file_processor/pypdf/pypdf.py (hunk 3 above)
Comment (Artemon-line) [reply src/ogx/providers/inline/file_processor/pypdf/pypdf.py:112]: Thanks! 🙂

**Comment by Artemon-line** [comment]
@subramen please take another look when you have a minute.
thanks

**Review by skamenan7 — APPROVED** [review]
